src/pkg/services/m365/api/user_info.go

```diff
@@ -0,0 +1,187 @@
+package api
+
+import (
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+
+	"github.com/alcionai/corso/src/internal/common/str"
+	"github.com/alcionai/corso/src/internal/common/tform"
+	"github.com/alcionai/corso/src/pkg/path"
+)
+
+// ---------------------------------------------------------------------------
+// User Info
+// ---------------------------------------------------------------------------
+
+type UserInfo struct {
+	ServicesEnabled map[path.ServiceType]struct{}
+	Mailbox         MailboxInfo
+}
+
+type MailboxInfo struct {
+	Purpose                    string
+	ArchiveFolder              string
+	DateFormat                 string
+	TimeFormat                 string
+	DelegateMeetMsgDeliveryOpt string
+	Timezone                   string
+	AutomaticRepliesSetting    AutomaticRepliesSettings
+	Language                   Language
+	WorkingHours               WorkingHours
+	ErrGetMailBoxSetting       []error
+	QuotaExceeded              bool
+}
+
+type AutomaticRepliesSettings struct {
+	ExternalAudience       string
+	ExternalReplyMessage   string
+	InternalReplyMessage   string
+	ScheduledEndDateTime   timeInfo
+	ScheduledStartDateTime timeInfo
+	Status                 string
+}
+
+type timeInfo struct {
+	DateTime string
+	Timezone string
+}
+
+type Language struct {
+	Locale      string
+	DisplayName string
+}
+
+type WorkingHours struct {
+	DaysOfWeek []string
+	StartTime  string
+	EndTime    string
+	TimeZone   struct {
+		Name string
+	}
+}
+
+func newUserInfo() *UserInfo {
+	return &UserInfo{
+		ServicesEnabled: map[path.ServiceType]struct{}{
+			path.ExchangeService: {},
+			path.OneDriveService: {},
+		},
+	}
+}
+
+// ServiceEnabled returns true if the UserInfo has an entry for the
+// service.  If no entry exists, the service is assumed to not be enabled.
+func (ui *UserInfo) ServiceEnabled(service path.ServiceType) bool {
+	if ui == nil || len(ui.ServicesEnabled) == 0 {
+		return false
+	}
+
+	_, ok := ui.ServicesEnabled[service]
+
+	return ok
+}
+
+// Returns if we can run delta queries on a mailbox. We cannot run
+// them if the mailbox is full which is indicated by QuotaExceeded.
+func (ui *UserInfo) CanMakeDeltaQueries() bool {
+	return !ui.Mailbox.QuotaExceeded
+}
+
+func parseMailboxSettings(
```

Comment:
We should add tests for this.

Reply:
It should already have test coverage from other PRs.